### DISCUSSION TITLE
Led 1 is blinking

### DISCUSSION THREAD

**Discussion by noammmmmm** [bitcraze/discussions#204] [Q&A]
Hello

When the CF is ok 
Power on and all is good: The blue LEDs (2 and 3) are fully lit and the front right LED (1) is blinking red twice every second.

But I get:
Power on and all is good but sensors are not yet calibrated: The blue LEDs (2 and 3) are fully lit and the front right LED (1) is blinking red with 2 seconds interval. Put the Crazyflie 2.X on a level surface and keep it absolutely still to calibrate.

What should I do to calibrate it?

Thanks
Noam

**Comment by knmcguire** [comment]
Hi!

Some additional questions:

- Is this a completely new crazyflie or an older one?
- Have you uploaded any modified firmware to it
- Have you looked at the raw sensor data of the imu in the cfclients plotter, perhaps compared it to another crazyflie?

**Comment by noammmmmm** [comment]
Another thing i notices

When i am trying to connect with the python, I recieved the message:
"ERROR:cflib.crazyflie.mem:-------->Got good count, but no info on mem!"
Attached my python file

Thanks
Noam
________________________________

**Comment by tobbeanton** [comment]
It could be the IMU (BMI088) that has become noisier for some reason. Please check [this](https://forum.bitcraze.io/viewtopic.php?p=23216&hilit=BMI088#p23216) old forum thread for some ideas.

**Comment by noammmmmm** [comment]
How can I know what is my IMU?
________________________________

**Reply by tobbeanton** [reply]
[IMU link to wikipedia](https://en.wikipedia.org/wiki/Inertial_measurement_unit)

Do you know how to compiled the firmware yourself? If so change the [GYRO_VARIACE_BASE](https://github.com/bitcraze/crazyflie-firmware/blob/master/src/hal/src/sensors_bmi088_bmp388.c#L85) from 100 to 200. Recompile, flash and test the new value.

**Comment by noammmmmm** [comment]
Hello
I
There is any  way please to get support with remote desktop?

Thanks
Noam
________________________________

**Reply by tobbeanton** [reply]
No sorry, we don't provide this

**Comment by noammmmmm** [comment]
Hello

Now it seems that the LED is blinking several times every second.


  *   Is this a completely new crazyflie or an older one?

I have this CF for a half of year. It was ok before.

  *   Have you uploaded any modified firmware to it
  *   Yes. I updated the firmware that i download from your site
  *
  *   Have you looked at the raw sensor data of the imu in the cfclients plotter, perhaps compared it to another crazyflie?

Do you mean the plotter tab in the cfclients? Well... there is no data.
But I see that sometimes when I turn on the CF it is communicating and sometimes not.
I didn't figure out when it happens.
Attached MP4 file
https://drive.google.com/file/d/1QkHbS-iNq-ydbif5kfH55dgHCoWJdXbK/view?usp=sharing
[https://lh3.googleusercontent.com/KfqxsYSiFNSGhYrBakj1TUYVaWl3jOV-_y5RZvRCzlDl9BBPHZ19CWeiHuj-zH1TpNQ=w1200-h630-p]<https://drive.google.com/file/d/1QkHbS-iNq-ydbif5kfH55dgHCoWJdXbK/view?usp=sharing>
WhatsApp Video 2022-08-20 at 19.46.15.mp4<https://drive.google.com/file/d/1QkHbS-iNq-ydbif5kfH55dgHCoWJdXbK/view?usp=sharing>
drive.google.com

I tried to update the last firmware from the VM and from my windows.

Thanks

Noam

________________________________

**Reply by krichardsson** [reply]
This blinking sequence means that something weird happened in the firmware and that it stopped. 
To investigate the problem, open the "Console" tab in the client (from the View/Tabs) menu. Restart the Crazyflie and connect to it in the client. You should now get some logs in the console tab when the Crazflie started and hopefully, some information of what the problem might be. Please post the logs here so we can take a look at it.

**Comment by noammmmmm** [comment]
This is what i get in the console:


SYS: ----------------------------

SYS: Crazyflie 2.1 is up and running!

SYS: Build 2:6ae2d411447d (2022.05 +2) MODIFIED

SYS: I am 0x383535303539470E002B0025 and I have 1024KB of flash!

CFGBLK: v1, verification [OK]

DECK_CORE: 1 deck(s) found

DECK_CORE: Calling INIT on driver bcFlow2 for deck 0

ZR2: Z-down sensor [OK]

PMW: Motion chip id: 0x48:0xB6

IMU: BMI088: Using I2C interface.

IMU: BMI088 Gyro connection [OK].

IMU: BMI088 Accel connection [OK]

IMU: BMP388 I2C connection [OK]

ESTIMATOR: Using Kalman (2) estimator

CONTROLLER: Using PID (1) controller

MTR-DRV: Using brushed motor driver

SYS: About to run tests in system.c.

SYS: NRF51 version: 2022.01*

EEPROM: I2C connection [OK].

STORAGE: Storage check [OK].

IMU: BMI088 gyro self-test [OK]

Error while initializing the PMW3901 sensor

DECK_CORE: Deck 0 test [FAIL].

SYS: deck [FAIL]

**Reply by krichardsson** [reply]
It looks like your flow deck is not initializing properly. There are two possible reasons:
1. The connection to the deck is bad. Check the connectors on the expansion port on the Crazyflie to make sure they look good.
2. The deck is broken. If you think your deck is broken, please email us at support@bitcraze.io for further assistance

**Comment by noammmmmm** [comment]
Hello

I bought a new deckflow and I still see the same problem.
Withoot the deck it seems that the problem gone.

SYS: Crazyflie 2.1 is up and running!

SYS: Build 98:2eb06560f24d (2022.01 +98) MODIFIED

SYS: I am 0x383535303539470E002B0025 and I have 1024KB of flash!

CFGBLK: v1, verification [OK]

DECK_CORE: 1 deck(s) found

DECK_CORE: Calling INIT on driver bcFlow2 for deck 0

ZR2: Z-down sensor [OK]

PMW: Motion chip id: 0x48:0xB6

IMU: BMI088: Using I2C interface.

IMU: BMI088 Gyro connection [OK].

IMU: BMI088 Accel connection [OK]

IMU: BMP388 I2C connection [OK]

ESTIMATOR: Using Kalman (2) estimator

CONTROLLER: Using PID (1) controller

MTR-DRV: Using brushed motor driver

SYS: About to run tests in system.c.

SYS: NRF51 version: 2022.01

EEPROM: I2C connection [OK].

STORAGE: Storage check [OK].

IMU: BMI088 gyro self-test [OK]

Error while initializing the PMW3901 sensor

DECK_CORE: Deck 0 test [FAIL].

SYS: deck [FAIL]


What can I do?

________________________________

**Reply by hmllr** [reply]
It seems as if you get a wrong id from the PMW sensor (the motion sensor on the flow deck) and therefor enter an error state. The id should be 0x49, the inverse id 0xB6, you get 0x48 and 0xB6 instead. I never saw this issue before (and those are read only registers, so it shouldn't happen that you accidentally overwrite them) - but it also seems as if you use a heavily modified firmware, so maybe try with the newest release?

**Comment by noammmmmm** [comment]
Can you send me please a link to the newest firmware?
________________________________

**Reply by hmllr** [reply]
https://github.com/bitcraze/crazyflie-firmware/tree/2022.12 this is the newest release - but you can also update it using the cfclient (https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/userguides/userguide_client/#firmware-upgrade)

**Comment by noammmmmm** [comment]
I have got this screen. No "Program" button is available.
________________________________

**Reply by hmllr** [reply]
Unfortunately it seems as if your screenshot didn't make it here, so I cannot tell what the issue is - you have to be connected to a Crazyflie and have chosen a firmware to update with (which then has to have finished being downloaded) to be able to program the Crazyflie.

**Comment by noammmmmm** [comment]
[cid:7584aa34-5563-4735-bf70-b55353ea0830]
________________________________
